ui/js/perf.js

```diff
@@ -659,3 +659,157 @@ perf.filter('displayPrecision', function() {
+    // error handler helper
+    var prevRevsErrHandler = function(response) {
+        if (response instanceof Error) {
+            return $q.reject("Unable to get previous revisions" +
+                             " (" + response + ").");
+        } else {
+            // http error
+            return $q.reject("Error " + response.status +
+                             " (" + response.data + ").");
+        }
+    };
+
+    /* *recursive* attempts to find the nearest branch and revision for a
+       revision that have a child present in another branch (e.g., find the
+       previous revision from try).
+
+     The logic is the following:
+
+     1. use `revision` and `project` to find the oldest changeset in the
+        push containing `revision`
+     2. find the parent of that revision
+     3. search in other `projects` the presence of that parent revision
+     4. If we find it, we are done. Else go to 1. using the parent revision
+        for the new `revision`.
+     */
+    var _getPreviousRevisionFrom = function(project, revision, projects,
+                                             defer, attempt, max_attempts) {
+        attempt +=1;
+        if (attempt >= max_attempts) {
+            defer.reject("Unable to find a previous revision");
+            return;
+        }
+
+        $http.get(
+            project.url + "/json-pushes?changeset=" + revision
+        ).then(function(response) {
+            // find the oldest changeset in the push
+            var pushId = _.keys(response.data)[0];
+            return response.data[pushId].changesets[0];
+        }).then(function(changeset) {
+            // find the parent changeset
+            return $http.get(
+                project.url + "/json-rev/" + changeset
+            ).then(function(response) {
+                return response.data.parents[0];
+            });
+        }).then(function(parentChset) {
+            // now check in projects if we can find the parent changeset
+            var promises = _.map(projects, function (proj) {
+                return $q(function(resolve, reject) {
+                    $http.get(
+                        proj.url + "/json-pushes?changeset=" + parentChset
+                    ).then(function(response) {
+                        var pushId = _.keys(response.data)[0];
+                        var chsets = response.data[pushId].changesets;
+                        return {
+                            revision: chsets[chsets.length - 1],
+                            date: response.data[pushId].date
+                        };
+                    }).then(
+                        // we found something!
+                        function(data) {
+                            resolve({
+                                project: proj,
+                                revision: data.revision,
+                                date: data.date
+                            });
+                        }, function(_) {
```

Comment:
I think what uo are talking about is done, https://github.com/mozilla/treeherder/pull/1283/files#diff-46f0e1fed9a8eba49973a0d54fb377e7R807.